EpiserverRedirects/Repository/DynamicDataStoreRepository.cs

```diff
@@ -10,7 +10,31 @@ public class DynamicDataStoreRepository : RedirectRuleRepository
         private readonly DynamicDataStoreFactory _dynamicDataStoreFactory;
         private DynamicDataStore DynamicDataStore => _dynamicDataStoreFactory.CreateStore(typeof(RedirectRule));
 
+        private IQueryable<RedirectRule> GetAllByOldPattern(string oldPattern)
+        {
+            return DynamicDataStore.Items<RedirectRule>().Where(x => x.OldPattern == oldPattern);
+        }
         
+        private IQueryable<RedirectRule> RemoveDuplicates(RedirectRule redirectRule)
+        {
+            var oldPatternRedirectRules = 
+                GetAllByOldPattern(redirectRule.OldPattern)
+                    .Where(x => x.RedirectRuleType != RedirectRuleType.Regex)
+                    .OrderByDescending(x => x.CreatedOn);
+            
+            IQueryable<RedirectRule> duplicates  = null;
+            
+            if (oldPatternRedirectRules.Count() > 1)
+            {
+                duplicates = oldPatternRedirectRules.Skip(1);
+
+                foreach (var duplicate in duplicates)
+                {
+                    Delete(duplicate.Id.ExternalId);
+                }
+            }
+            return duplicates;
```

Comment:
Since this method returns `IQueryable<RedirectRule>` (which I'm not 100% sure is required, since it is not used anywhere) I think to be on safe side we should check for null here so the last line:
`return duplicates ?? Enumerable.Empty<RedirectRule>().AsQueryable();`
so we do not hit NullReferenceException by accident